content/en/docs/concepts/security/security-checklist.md

```diff
@@ -0,0 +1,154 @@
+---
+reviewers:
+title: Security Checklist
+description: >
+  Recommended checklist for ensuring security in Kubernetes clusters.
+content_type: concept
+---
+
+<!-- overview -->
+
+This checklist aims at providing a basic list of guidance with links to more
+comprehensive documentation on each topic. It does not claim to be exhaustive
+and is meant to evolve.
+
+The order of topics does not reflect an order of priority.
+
+<!-- body -->
+
+## Authentication & Authorization
+
+See the [Role Based Access Control Good Practices](/docs/concepts/security/rbac-good-practices/)
+for information related to authentication and authorization.
+
+<!-- New authentication certificates should be generated on new clusters that avoid -->
+<!-- the use of the `system:masters`. `system:masters` should be used as a -->
+<!-- break-glass mechanism as opposed to an admin user. -->
+
+## Network security
+
+A number of [Container Network Interface (CNI) plugins](/docs/concepts/extend-kubernetes/compute-storage-net/network-plugins/)
+plugins provide the functionality to
+restrict network resources pods may communicate with. This is most commonly done
+through [Network Policies](/docs/concepts/services-networking/network-policies/)
+which provide a namespaced resource to define rules.
+
+Not all CNI plugins provide encryption in transit. If the chosen plugin lacks this
+feature, an alternative solution could be to use a service mesh to provide that
+functionality.
+
+The `etcd` datastore of the control plane should have controls to limit access.
+Furthermore, mutual TLS (mTLS) should be used to communicate securely with it.
+The certificate authority for this should be unique to `etcd`.
+In addition, external Internet access to the Kubernetes API server should be restricted
+to not expose the API publicly. You can then use a bastion host to access the server.
+If a cloud provider is used for hosting Kubernetes, the access from pods to the cloud
+metadata API `169.254.169.254` should also be restricted or blocked if not needed
+because it may leak information.
+
+## Pod security
+
+RBAC authorization is crucial but
+[cannot be specific enough on the Pods resource](/docs/concepts/security/rbac-good-practices/#workload-creation)
+(or on any resource that manages Pods). The only granularity is the API verbs
+on the resource itself, for example `create` on Pods. Indeed, without
+additional admission, the authorization to create these resources allows direct
+unrestricted access to on the schedulable nodes of a cluster.
+
+[Pod Security Admission](/docs/concepts/security/pod-security-admission/) is an
```

Comment:
Is it worth adding something that says a webhook admission controller could be used should finer grain control compared to pod security admission be required? Thinking of things like OPA, Kyverno, etc.. of course it doesn't need ot mention them by name

Reply:
For a checklist, I think people could pick from a menu:
- Pod Security admission only
- webhook admission control and external service, only
- webhook admission control and external service _plus_ Pod Security admission

This can be two different checklist-type items:
1. I have one of the above in place
1. I have evaluated my choice from that menu against my information security needs

Reply:
> webhook admission control and external service _plus_ Pod Security admission

This feels like good defence in depth: if you break your OPA config, you can still enforce the baseline Pod Security Standard for all the namespaces you labelled.

Reply:
I tried to clarify that here https://github.com/kubernetes/website/pull/33992/commits/eb962b4c124212b0c3cf2a01a93aba9bc251cb7c.